libeos-updater-util/checkpoint.c

```diff
@@ -0,0 +1,145 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright 2024 Endless OS Foundation LLC
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include <libeos-updater-util/checkpoint-private.h>
+#include <libeos-updater-util/enums.h>
+#include <glib/gi18n.h>
+
+G_DEFINE_QUARK (euu-checkpoint-block-quark, euu_checkpoint_block)
+
+/**
+ * euu_checkpoint_block_to_string:
+ * @reason: An element of #EuuCheckpointBlock
+ *
+ * Helper function to get the nickname of @reason.
+ *
+ * Returns: the nickname of @reason, such as "forced"
+ */
+const char *
+euu_checkpoint_block_to_string (EuuCheckpointBlock reason)
+{
+  g_autoptr(GEnumClass) enum_class = G_ENUM_CLASS (g_type_class_ref (EUU_TYPE_CHECKPOINT_BLOCK));
+  GEnumValue *value = g_enum_get_value (enum_class, (gint) reason);
+
+  g_return_val_if_fail (value != NULL, NULL);
+  return value->value_nick;
+}
+
+static gboolean
+is_nvme_remap_in_use (OstreeSysroot *sysroot)
```

Comment:
I don't actually have the relevant hardware. I will build some packages and ask those in Taipei to check them, if this doesn't land before that.

Reply:
@starnight confirmed that this works correctly on the device he has that uses `nvme-remap`. Thanks!